packages/core/lib/commands/develop/run.js

```diff
@@ -28,7 +29,7 @@ module.exports = async options => {
   const customConfig = config.networks.develop || {};
 
   const { mnemonic, accounts, privateKeys } = mnemonicInfo.getAccountsInfo(
-    customConfig.accounts || 10
+    customConfig.accounts || customConfig.total_accounts || 10
```

Comment:
Aha, the synonyms! :)

One note here -- and I realize that this problem is pre-existing and not due to your changes, but as long as you're getting the rest of this, you might want to handle this here -- this code means that `accounts` is zero, it'll end up as 10 instead.  Perhaps this should be changed so that an input of `0` will be respected?  (There isn't some problem with this, is there @davidmurdoch?)

Reply:
I am not very sure of this. But in my opinion if `accounts` is `0`, then the commands like `test` and `migrate` won't work as those commands need at least one account to work as expected.

Reply:
As of 7.0 ganache _allows_ 0 accounts on startup, and users can then add others later via RPC. They can also start ganache with unlocked accounts, though I don't know if truffle allows that option to be passed through from truffle config to ganache.

Reply:
> I am not very sure of this. But in my opinion if accounts is 0, then the commands like test and migrate won't work as those commands need at least one account to work as expected.

I mean, so what?  To my mind, if the user requests an operation that will error, they should get the operation and error they requested!

Reply:
Okay, sure. I agree with you @haltman-at . We should put checks for these and throw the errors as required.

Reply:
Oh, I wasn't suggesting putting a check... I was suggesting we just let the consequences play out if that's what the user really wants. :)  After all, Truffle Develop and Truffle Test don't necessarily *require* accounts, right?  They *usually* do, but not always, so I can imagine someone wanting to set it to 0...

But maybe a check is a good idea...?  IDK, @gnidan, any opinion here?

Reply:
I was thinking of throwing appropriate errors for each case (`accounts`, `defaultEtherBalance` and `gasPrice`) so that user becomes aware of the managed ganache network requirements. Any suggestions on this @cds-amal @eggplantzzz @davidmurdoch ?

Reply:
Is it always true that they won't work, though?

I'm not sure what the effect of `gasPrice: 0` would be.  Does it just mean gas is free?

Setting `defaultEtherBalance` to 0 seems like a problem... but if it's OK to also set `gasPrice` to 0, so that gas is free, then it doesn't matter if you also have no ether, right? :)

Similarly, `accounts: 0` means you can't use the unlocked accounts... but there's nothing stopping you from making your own private key and making your own account.  Aside perhaps from a lack of ether, but I imagine that can be addressed either with `evm_mine` calls, or, again, by making gas free so that ether is unnecessary.

These may be fringe use cases, but I don't see a reason to bar them.  Unless `gasPrice: 0` doesn't actually make gas free, anyway.

Reply:
I have tested with `gasPrice: 0` which seems to make no difference. `migrate` and `test` works with only `gasPrice: 0`. And if `gasPrice` and `defaultBalanceEther` are both 0, then the transaction fails with **insufficient funds**. May be that means `gasPrice: 0` is not making gas free.

Reply:
Oh, interesting...